GoogleDataTransport/GDTCORLibrary/GDTCORUploadCoordinator.m

```diff
@@ -141,7 +141,25 @@ - (void)uploadTargets:(NSArray<NSNumber *> *)targets conditions:(GDTCORUploadCon
  */
 - (GDTCORUploadConditions)uploadConditions {
 #if TARGET_OS_WATCH
-  return GDTCORUploadConditionNoNetwork;
+  dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+  NSURL *url = [NSURL URLWithString:@"https://google.com"];
+  NSURLSession *currentURLSession = [NSURLSession sharedSession];
+  BOOL __block reachablity = NO;
+  NSURLSessionTask *networkTask =
+      [currentURLSession dataTaskWithURL:url
+                       completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                         if (!error && response) {
+                           reachablity = YES;
+                         }
+                         dispatch_semaphore_signal(semaphore);
```

Comment:
For watchOS, we don't want to send a request and wait for the response to get upload conditions--this would be very slow and would consume a lot of battery. We should just assume that the network is there and then gracefully handle the response or lack thereof. I'd add another enum to GDTCORUploadConditions that's called something like `GDTCORUploadConditionAssumedConnected` and just set it to that for watchOS.